src/http_rest_api_handler.cpp

```diff
@@ -60,39 +64,92 @@ Status HttpRestApiHandler::parseModelVersion(std::string& model_version_str, std
             return processPredictRequest(request_components.model_name, request_components.model_version,
-                request_components.model_version_label, request_body, response);
+                request_components.model_version_label, request_body, &response);
         } else {
             SPDLOG_WARN("Requested REST resource not found");
-            return StatusCode::REST_NOT_FOUND;
+            return (Status)StatusCode::REST_NOT_FOUND;
         }
-    }
-    if (request_components.type == GetModelMetadata) {
+    });
+
+    registerHandler(GetModelMetadata, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         return processModelMetadataRequest(request_components.model_name, request_components.model_version,
-            request_components.model_version_label, response);
-    }
-    if (request_components.type == GetModelStatus) {
+            request_components.model_version_label, &response);
+    });
+    registerHandler(GetModelStatus, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         return processModelStatusRequest(request_components.model_name, request_components.model_version,
-            request_components.model_version_label, response);
-    }
-    if (request_components.type == ConfigReload) {
+            request_components.model_version_label, &response);
+    });
+    registerHandler(ConfigReload, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         auto& manager = ModelManager::getInstance();
-        return processConfigReloadRequest(*response, manager);
-    }
-    if (request_components.type == ConfigStatus) {
+        return processConfigReloadRequest(response, manager);
+    });
+    registerHandler(ConfigStatus, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         auto& manager = ModelManager::getInstance();
-        return processConfigStatusRequest(*response, manager);
+        return processConfigStatusRequest(response, manager);
+    });
+    registerHandler(KFS_GetModelStatus, processModelStatusKFSRequest);
+    registerHandler(KFS_GetModelMetadata, processModelMetadataKFSRequest);
+}
+
+
+Status HttpRestApiHandler::dispatchToProcessor(
+    const std::string& request_body,
+    std::string* response,
+    const HttpRequestComponents& request_components) {
+
+
+    if(handlers.count(request_components.type) > 0){
+        return handlers[request_components.type](request_components, *response, request_body);
+    }else{
+        return StatusCode::UNKNOWN_REQUEST_COMPONENTS_TYPE;
     }
+
     return StatusCode::UNKNOWN_REQUEST_COMPONENTS_TYPE;
 }
 
+Status HttpRestApiHandler::processModelStatusKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
```

Comment:
Rename *KFS*ModelStatus* occurences to KFS*ModelReady* to beter reflect the API.

TFS has Model Status and KFS Model Ready is equivalent of that. In code it's better to follow the API naming